packages/core/src/fhir-deduplication/resources/observation.ts

```diff
@@ -104,3 +108,16 @@ export function groupSameObservations(observations: Observation[]): {
     danglingReferences: [...danglingReferencesSet],
   };
 }
+
+function preProcess(observation: Observation): { observation: Observation; code: CodeableConcept } {
+  const newObservation = { ...observation };
```

Comment:
This does a shallow copy - i.e., if we change a second level property that will do so on the original object.

To be safe, I'd use Lodash's `cloneDeep`.